packages/micropip/test_micropip.py

```diff
@@ -61,24 +59,34 @@ def test_parse_wheel_url():
 
 def test_install_custom_url(selenium_standalone, web_server_tst_data):
     server_hostname, server_port, server_log = web_server_tst_data
-    selenium_standalone.load_package("micropip")
-    selenium_standalone.run("import micropip")
     base_url = f"http://{server_hostname}:{server_port}/"
     url = base_url + "snowballstemmer-2.0.0-py2.py3-none-any.whl"
-    selenium_standalone.run(f"micropip.install('{url}')")
-    # wait untill micropip is loaded
-    time.sleep(1)
-    selenium_standalone.run("import snowballstemmer")
+    selenium_standalone.run_js(
+        f"""
+        await pyodide.runPythonAsync(`
+            import micropip
+            await micropip.install('{url}')
+            import snowballstemmer
+        `);
+        """
+    )
 
 
 def test_add_requirement_relative_url():
     pytest.importorskip("distlib")
     import micropip
 
     transaction = {"wheels": []}
-    micropip.PACKAGE_MANAGER.add_requirement(
+    coroutine = micropip.PACKAGE_MANAGER.add_requirement(
         "./snowballstemmer-2.0.0-py2.py3-none-any.whl", {}, transaction
     )
+    try:
+        coroutine.send(None)
```

Comment:
I can find very little documentation on this use case. Could you add a comment that it's a way to run that coroutine from a sync function?

Reply:
Yeah I agree this is hardly described anywhere. It's kind of surprising to me that there isn't more explanation for this, because all of the `EventLoop` and asyncio in Python is abstractions on top of this: a coroutine is just a generator, whenever it wants to do io, etc it yields a request and the event loop schedules the request and when it gets the result `send`s it back to the coroutine, when it's done it raises `StopIteration`. If the coroutine does no async operations, then it just runs to completion on the first try. I can add a comment. Using `pytest-asyncio` would probably be easier to understand though.